### PR TITLE
Fixes #1079 Tablet: Lowfi Onboarding Flow (landscape)

### LINKED ISSUE #1079: Onboarding Flow - Tablet (Landscape) (Lowfi)
Mock: https://xd.adobe.com/view/d405de00-a871-4f0f-73a0-f8acef30349b-a234/

Implement low-fi UI for **Onboarding Flow** tablet landscape mode

**Target PR date**: 3 June 2020
**Target completion date**: 6 June 2020

### PR DESCRIPTION
**Pull request by MohamedMedhat1998**
## Explanation
Fixes #1079  
Created landscape-tablet layouts for onboarding flow (low-fi). These layouts have the qualifier "sw600dp" for tablets, combined with the qualifier "land" for landscape mode

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: app/src/main/res/layout-sw600dp-land/onboarding_slide_final.xml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      app:layout_constraintTop_toBottomOf="@id/slide_description_text_view" />
+
+    <androidx.constraintlayout.widget.Guideline
+      android:id="@+id/guideline2"
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:orientation="horizontal"
+      app:layout_constraintGuide_percent="0.5" />
+  </androidx.constraintlayout.widget.ConstraintLayout>
+</layout>
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-land/onboarding_slide_final.xml]: EOF is missing, add one blank line at the end.

File: app/src/main/res/layout-sw600dp-land/onboarding_slide.xml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      app:layout_constraintTop_toTopOf="@+id/guideline" />
+
+    <androidx.constraintlayout.widget.Guideline
+      android:id="@+id/guideline"
+      android:layout_width="wrap_content"
+      android:layout_height="wrap_content"
+      android:orientation="horizontal"
+      app:layout_constraintGuide_percent="0.5" />
+  </androidx.constraintlayout.widget.ConstraintLayout>
+</layout>
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-land/onboarding_slide.xml]: EOF is missing, add one blank line at the end.

File: app/src/main/res/layout-sw600dp-land/onboarding_fragment.xml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+          android:textAllCaps="true"
+          android:textColor="@color/oppiaPrimaryText"
+          android:textSize="14sp"
+          android:visibility="@{viewModel.slideNumber != (viewModel.totalNumberOfSlides - 1) ? View.VISIBLE: View.GONE, default=visible}"
+          app:layout_constraintStart_toStartOf="parent"
+          app:layout_constraintTop_toTopOf="@+id/slide_dots_container" />
+      </androidx.constraintlayout.widget.ConstraintLayout>
+    </FrameLayout>
+  </androidx.constraintlayout.widget.ConstraintLayout>
+</layout>
```
Comment (rt4914) [inline app/src/main/res/layout-sw600dp-land/onboarding_fragment.xml]: EOF is missing, add one blank line at the end.

**Review by rt4914 — CHANGES_REQUESTED** [review]
In low-fi always check the test cases. `testOnboardingFragment_checkSlide3Image_isCorrect` test case is failing because some of the xml files does not have a content description.

If you are aware of this test failure, in such cases please make sure that you mention this in description of PR so that the reviewer knows it.

**Comment by rt4914** [comment]
@MohamedMedhat1998 Please fix the test case issue and also, reply to all conversations.

File: app/src/main/res/layout-sw600dp-land/onboarding_slide_final.xml (hunk 1 above)
Comment (MohamedMedhat1998) [reply app/src/main/res/layout-sw600dp-land/onboarding_slide_final.xml]: Done

File: app/src/main/res/layout-sw600dp-land/onboarding_slide.xml (hunk 2 above)
Comment (MohamedMedhat1998) [reply app/src/main/res/layout-sw600dp-land/onboarding_slide.xml]: Updated

File: app/src/main/res/layout-sw600dp-land/onboarding_fragment.xml (hunk 3 above)
Comment (MohamedMedhat1998) [reply app/src/main/res/layout-sw600dp-land/onboarding_fragment.xml]: Done

**Comment by MohamedMedhat1998** [comment]
> 
> 
> In low-fi always check the test cases. `testOnboardingFragment_checkSlide3Image_isCorrect` test case is failing because some of the xml files does not have a content description.
> 
> If you are aware of this test failure, in such cases please make sure that you mention this in description of PR so that the reviewer knows it.

Done

**Comment by MohamedMedhat1998** [comment]
> 
> 
> @MohamedMedhat1998 Please fix the test case issue and also, reply to all conversations.

Completed!  
Can you review it again please?

**Comment by rt4914** [comment]
> > @MohamedMedhat1998 Please fix the test case issue and also, reply to all conversations.
> 
> Completed!
> Can you review it again please?

@MohamedMedhat1998 Just add `android:contentDescription="@string/onboarding_slide_3_title"` to `layout_land/onboarding_slide_final.xml` `ImageView`. 
Checkout its portrait mode code fort comparision.

**Comment by MohamedMedhat1998** [comment]
> 
> 
> > > @MohamedMedhat1998 Please fix the test case issue and also, reply to all conversations.
> > 
> > 
> > Completed!
> > Can you review it again please?
> 
> @MohamedMedhat1998 Just add `android:contentDescription="@string/onboarding_slide_3_title"` to `layout_land/onboarding_slide_final.xml` `ImageView`.
> Checkout its portrait mode code fort comparision.

Done!  
Added `contentDescription` in normal devices and in tablet devices

**Review by rt4914 — APPROVED** [review]
LGTM. Thanks.

**Comment by rt4914** [comment]
@vinitamurthi @veena14cs Can you have a look at this. The test cases are failing for `SplashActivityTest` (non_flaky_tests).
When I run this file on espresso, I am getting this error:
`java.lang.RuntimeException: Delegate runner 'org.robolectric.RobolectricTestRunner' for AndroidJUnit4 could not be loaded.`

I tried to solve this error using [this link](https://stackoverflow.com/questions/53619024/attempted-to-use-androidjunit4-with-standard-junit-runner-and-delegate-runner-o), but everything is correct as per the link.

**Comment by vinitamurthi** [comment]
As veena mentioned in the chat, it looks like it is working on develop. Does this PR affect the splash activity in any way?

**Comment by rt4914** [comment]
> As veena mentioned in the chat, it looks like it is working on develop. Does this PR affect the splash activity in any way?

@veena14cs @vinitamurthi 
Solved the issue. Actually it was quite weird issue. On develop they were passing but on this PR branch they were failing. Actually on PR branch I tried running other test cases also on espresso and all test cases were failing. So I did gradle sync again and it worked. Somehow the library was working for develop but not for this branch.

After gradle sync everything worked.

There is still on issue, why does is fail on CircleCI? So for that I have restarted the CircleCI job now and lets see what the result is.

**Comment by MohamedMedhat1998** [comment]
Can I merge now?

**Comment by rt4914** [comment]
> > As veena mentioned in the chat, it looks like it is working on develop. Does this PR affect the splash activity in any way?
> 
> @veena14cs @vinitamurthi
> Solved the issue. Actually it was quite weird issue. On develop they were passing but on this PR branch they were failing. Actually on PR branch I tried running other test cases also on espresso and all test cases were failing. So I did gradle sync again and it worked. Somehow the library was working for develop but not for this branch.
> 
> After gradle sync everything worked.
> 
> There is still on issue, why does is fail on CircleCI? So for that I have restarted the CircleCI job now and lets see what the result is.

@veena14cs @vinitamurthi The test cases are passing now.

**Pull request merged**